src/database/database-internal.ts

```diff
@@ -96,9 +106,43 @@ export class DatabaseService {
 
       this.databases[dbUrl] = db;
     }
+
+    if (!this.tokenRefresh) {
+      this.tokenRefresh = true;
+      this.appInternal.INTERNAL.addAuthTokenListener(this.onTokenChange);
+    }
+
     return db;
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  private onTokenChange(_: string): void {
+    this.appInternal.INTERNAL.getToken()
+      .then((token) => {
+        const delayMillis = token.expirationTime - TOKEN_REFRESH_THRESHOLD_MILLIS - Date.now();
+        // If the new token is set to expire soon (unlikely), do nothing. Somebody will eventually
+        // notice and refresh the token, at which point this callback will fire again.
+        if (delayMillis > 0) {
+          this.scheduleTokenRefresh(delayMillis);
+        }
+      })
+      .catch((err) => {
+        console.error('Unexpected error while attempting to schedule a token refresh:', err);
```

Comment:
It think the error most likely occurred during `getToken()` and not during the scheduling step.

Reply:
True. But overall this function is attempting to schedule a new token refresh event. It calls `getToken()` just to figure out when to schedule that event.